core/src/Cuda/Kokkos_Cuda_Half.hpp

```diff
@@ -943,6 +1092,25 @@ KOKKOS_INLINE_FUNCTION
 }
 #endif
 }  // namespace Experimental
+
+// use float as the return type for sum and prod since cuda_fp16.h
+// has no constexpr functions for casting to __half
+template <>
+struct reduction_identity<Kokkos::Experimental::half_t> {
+  KOKKOS_FORCEINLINE_FUNCTION constexpr static float sum() noexcept {
+    return 0.0F;
+  }
+  KOKKOS_FORCEINLINE_FUNCTION constexpr static float prod() noexcept {
+    return 1.0F;
+  }
+  KOKKOS_FORCEINLINE_FUNCTION constexpr static float max() noexcept {
+    return -65504.0F;
```

Comment:
Does CUDA have `std::numeric_limits<__half>::max()/min()` available?

Reply:
@maxpkatz this, too

Reply:
We have `Kokkos::Experimental::finite_min<float>::value` to replace std::numeric_limits in device code (but nothing for `__half`), also see https://stackoverflow.com/questions/24498140/is-there-a-cuda-equivalent-to-stdnumeric-limits.

Reply:
I will submit a request for `std::numeric_limits<__half>`. As indicated in the above SO post, CUDA doesn't provide numeric_limits in device code in general, however because these are constexpr functions, `--expt-relaxed-constexpr` generally works for providing this, so if we had a host version of this available it would likely be able to work in device code.